contracts/specs/NFTSwapper/NFTSwapperSpec.md

```diff
@@ -0,0 +1,163 @@
+# NFTSwapper Spec
+
+NFTSwapper enables permissionless swapping between ERC721's within a collection. 
+
+This contract is immutable with no owner. No NFTs or ETH are ever held by this contract, instead it just facilitates the permissionless swap process. 
+
+---
+
+- [NFTSwapper Spec](#nftswapper-spec)
+  - [Function Interface](#function-interface)
+    - [`constructor`](#constructor)
+    - [`swapPrice`](#swapprice)
+    - [`swap`](#swap)
+  - [Swap Price Curve](#swap-price-curve)
+    - [Constant Price Curve](#constant-price-curve)
+    - [Output NFT age based curve](#output-nft-age-based-curve)
+    - [Input and Output NFT age difference based curve](#input-and-output-nft-age-difference-based-curve)
+  - [Approval Considerations](#approval-considerations)
+
+## Function Interface
+
+### `constructor`
+
+```solidity
+constructor(
+    address swapPool, 
+    address nft, 
+    address feeRecipient, 
+    PriceCurveParams priceCurveParams
+)
+```
+
+* `swapPool`: the address which holds the swappable nft's, note that the `swapPool` must approve this contract for it's NFT's (or a subset of them, see [Approval Considerations](#approval-considerations)).
+* `nft`: the address of the ERC721 token that is being swapped.
+* `feeRecipient`: the address where swap fees will be sent. 
+* `priceCurveParams`: the parameters for the price curve (see [Swap Price Curve](#swap-price-curve)).
```

Comment:
`swapPool`, `nft`, `feeRecipient` meant to make this a bit more generic so it could be used by other collections / pools?

Reply:
Yea, exactly. Just going to set immutable params in constructor. 

In our case:
* `swapPool`: Nouns timelock
* `nft`: Nouns token
* `feeRecipient `: Nouns timelock